src/main/java/org/kiwiproject/dropwizard/util/admin/StandardAdminConfigurator.java

```diff
@@ -0,0 +1,117 @@
+package org.kiwiproject.dropwizard.util.admin;
+
+import static java.util.Objects.isNull;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotNull;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.google.common.annotations.VisibleForTesting;
+import io.dropwizard.setup.Environment;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import org.kiwiproject.config.TlsContextConfiguration;
+import org.kiwiproject.dropwizard.util.config.KeystoreConfig;
+import org.kiwiproject.dropwizard.util.health.HttpConnectionsHealthCheck;
+import org.kiwiproject.dropwizard.util.health.keystore.ExpiringKeystoreHealthCheck;
+import org.kiwiproject.dropwizard.util.metrics.ServerLoadGauge;
+import org.kiwiproject.dropwizard.util.task.ServerLoadTask;
+
+import java.util.stream.Stream;
+
+/**
+ * Set of utilities to assist in configuring administrative functions of a Dropwizard service.
+ */
+public class StandardAdminConfigurator {
+
+    private final Environment environment;
+    private TlsContextConfiguration tlsConfiguration;
+
+    @Getter(AccessLevel.PACKAGE)
+    @VisibleForTesting
+    private final Double leasedWarningThreshold;
+
+    @Getter(AccessLevel.PACKAGE)
+    @VisibleForTesting
+    private final boolean registerExpiringKeystoreHealthCheck;
+
+    @Getter(AccessLevel.PACKAGE)
+    @VisibleForTesting
+    private final boolean registerConfigResource;
+
+    @Builder
+    private StandardAdminConfigurator(Environment environment,
+                                      Double leasedWarningThreshold,
+                                      TlsContextConfiguration tlsConfiguration,
+                                      Boolean registerExpiringKeystoreHealthCheck,
+                                      Boolean registerConfigResource) {
+
+        this.environment = requireNotNull(environment, "Environment is required");
+        this.leasedWarningThreshold = isNull(leasedWarningThreshold) ? HttpConnectionsHealthCheck.DEFAULT_WARNING_THRESHOLD : leasedWarningThreshold;
+        this.registerExpiringKeystoreHealthCheck = isNull(registerExpiringKeystoreHealthCheck) || registerExpiringKeystoreHealthCheck;
+        this.registerConfigResource = isNull(registerConfigResource) || registerConfigResource;
+
+        if (this.registerExpiringKeystoreHealthCheck) {
+            this.tlsConfiguration = requireNotNull(tlsConfiguration,
+                    "TlsConfiguration is required when registering ExpiringKeystoreHealthCheck");
+        }
+    }
+
+    /**
+     * Sets up a task to retrieve the server load for the machine the service is running on.
```

Comment:
Should we make this a more generic message like `addHealthChecks` does, e.g. "Adds several default tasks..."

Reply:
Rewrote all of this